### ISSUE TITLE
Test fails when run with Scala 2.12.0

### ISSUE THREAD

**Issue by NthPortal** [NthPortal/app-shell#23]
The following test fails when run with Scala 2.12.0, but not 2.12.1

```
AsyncShellTest:
CancellableInputProvider
- should fail `nextAction` if it has been cancelled
```

**Comment by NthPortal** [comment]
See branch [build](https://github.com/NthPortal/app-shell/tree/be89689370e3425a591cf17f0e07d48d9b0747d0)

**Comment by NthPortal** [comment]
Caused by [this issue](https://issues.scala-lang.org/browse/SI-10034), which was fixed in 2.12.1.
